### PR TITLE
Print buffer

### PR DESCRIPTION
**Pull request by Mrokkk**

### CODE REVIEW

File: src/PrintBuffer.hpp [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        }
+        return {};
+    }
+
+    bool empty() const {
+        return position_ == buffer_.end();
+    }
+
+    bool operator==(const char *a) const {
+        return buffer_ == a;
```
Comment (szborows) [inline src/PrintBuffer.hpp]: is this safe (like strcmp vs strncmp)?

File: src/PrintBuffer.hpp [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+namespace yb {
+
+class PrintBuffer final {
+
+    using Container = std::string;
+    using Iterator = std::string::const_iterator;
+    Container buffer_;
+    Iterator position_;
+    const size_t defaultBufferSize_ = 1024;
```
Comment (szborows) [inline src/PrintBuffer.hpp]: constexpr

File: src/PrintBuffer.hpp [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        buffer_.reserve(defaultBufferSize_);
+        position_ = buffer_.end();
+    }
+
+    explicit PrintBuffer(size_t size) {
+        buffer_.reserve(size);
+        position_ = buffer_.end();
+    }
+
+    PrintBuffer &operator =(std::string &&s) {
```
Comment (szborows) [inline src/PrintBuffer.hpp]: are you sure we should use `operator=` here? it is quite close to move assignment operator and I guess it wasn't your intention?

File: src/PrintBuffer.hpp (hunk 1 above)
Comment (Mrokkk) [inline src/PrintBuffer.hpp]: Why it shouldn't be? I mean, std::string has a comparison operator which takes a `const char *` as a parameter.

File: src/PrintBuffer.hpp (hunk 2 above)
Comment (szborows) [inline src/PrintBuffer.hpp]: you're right, sorry

File: src/PrintBuffer.hpp (hunk 1 above)
Comment (szborows) [inline src/PrintBuffer.hpp]: apparently i'm just exaggerating :)

**Pull request merged**